src/core/smack-setup.c

```diff
@@ -264,6 +264,50 @@ static int write_netlabel_rules(const char* srcdir) {
         return r;
 }
 
+static int write_onlycap_list(void) {
+        _cleanup_close_ int onlycap_fd = -1;
+        _cleanup_free_ char *list = NULL;
+        _cleanup_fclose_ FILE *f = NULL;
+        size_t len = 0, allocated = 0;
+        char buf[LINE_MAX];
+        int r;
+
+        f = fopen("/etc/smack/onlycap", "re");
+        if (!f) {
+                if (errno != ENOENT)
+                        log_warning("Failed to read '/etc/smack/onlycap': %m");
+                return errno == ENOENT ? ENOENT : -errno;
+        }
+
+        FOREACH_LINE(buf, f, return -errno) {
+                if (isempty(truncate_nl(buf)) || strchr(COMMENTS, *buf))
+                        continue;
+
+                if (!GREEDY_REALLOC(list, allocated, len + strlen(buf) + 1))
+                        return -ENOMEM;
+
+                snprintf(list + len, allocated, "%s ", buf);
```

Comment:
please do not use snprintf() when all you want to do is copy some string. sprintf() and friends are relatively slow, and memcpy() is way faster. Maybe write this like this:

stpcpy(list + len, buf)[0] = ' ';